solutions/src/test/java/ru/sberbank/school/task10/ConcurrentFixedThreadPoolTest.java

```diff
@@ -0,0 +1,147 @@
+        assertEquals(core, nowThreadSize - initialThreadSize);
+
+        CountDownLatch latch = new CountDownLatch(25);
+        for (int i = 0; i < 25; i++) {
+            threadPool.execute(newRunnable50ms(counter, latch));
+        }
+        latch.await();
+        assertEquals(25, counter.get());
+
+        Thread.sleep(10);
+        nowThreadSize = Thread.getAllStackTraces().keySet().size();
+        assertEquals(core, nowThreadSize - initialThreadSize);
+
+        threadPool.stopNow();
+        Thread.sleep(10);
+        nowThreadSize = Thread.getAllStackTraces().keySet().size();
+        assertEquals(nowThreadSize, initialThreadSize);
+    }
+
+    @Test(timeout = 500)
+    public void callableService() throws InterruptedException, ExecutionException {
+        final List<Future<String>> futureList = new ArrayList<>();
+        final AtomicInteger counter = new AtomicInteger(0);
+        int initialThreadSize = Thread.getAllStackTraces().keySet().size();
+        int nowThreadSize;
+
+        threadPool.start();
+        Thread.sleep(10);
+        nowThreadSize = Thread.getAllStackTraces().keySet().size();
+        assertEquals(core, nowThreadSize - initialThreadSize);
+
+        CountDownLatch latch = new CountDownLatch(25);
+        for (int i = 0; i < 25; i++) {
+            Future<String> future = threadPool.execute(newCallable50ms(counter, latch));
+            futureList.add(future);
+        }
+        for (Future future : futureList) {
+            System.out.println(future.get());
+        }
+        latch.await();
+        assertEquals(25, counter.get());
+
+        Thread.sleep(10);
+        nowThreadSize = Thread.getAllStackTraces().keySet().size();
+        assertEquals(core, nowThreadSize - initialThreadSize);
+
+        threadPool.stopNow();
+        Thread.sleep(10);
+        nowThreadSize = Thread.getAllStackTraces().keySet().size();
+        assertEquals(nowThreadSize, initialThreadSize);
+    }
+
+    @Test
+    public void executeAfterStop() {
+        threadPool.start();
+        threadPool.stopNow();
+        Assertions.assertThrows(IllegalStateException.class, () ->
+                threadPool.execute(() -> {
+                }));
+    }
+
+    @Test
+    public void doubleStop() {
+        threadPool.start();
+        threadPool.stopNow();
+        Assertions.assertThrows(IllegalStateException.class, () -> threadPool.stopNow());
+    }
+
+    @Test
+    public void doubleStart() {
+        threadPool.start();
+        Assertions.assertThrows(IllegalStateException.class, () -> threadPool.start());
+
+    }
+
+    @Test
+    public void wrongThreadsCount() {
+        Assertions.assertAll(
+                () -> Assertions.assertThrows(IllegalArgumentException.class
+                        , () -> new ConcurrentFixedThreadPool(0)),
+                () -> Assertions.assertThrows(IllegalArgumentException.class
+                        , () -> new ConcurrentFixedThreadPool(-18))
```

Comment:
&#39;,&#39; should be on the previous line.

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a998197075>*